src/sherpa_ai/utils.py

```diff
@@ -325,3 +330,217 @@ def check_if_number_exist(result: str, source: str):
+
+    json_pattern = r"\{.*\}"
+    json_match = re.search(json_pattern, text)
+
+    if json_match:
+        json_data = json_match.group()
+        try:
+            parsed_json = json.loads(json_data)
+            return parsed_json
+        except json.JSONDecodeError as e:
+            return {}
+    else:
+        return {}
+
+
+def text_similarity_by_llm(
+    source_entity: List[str],
+    source,
+    result,
+    user_id=None,
+    team_id=None,
+):
+    """
+        Check if entities from a question are mentioned in some form inside the answer using a language model.
+
+        Args:
+        - source_entity (List[str]): List of entities from the question.
+        - source (str): Question text.
+        - result (str): Answer text.
+        - user_id (str): User ID (optional).
+        - team_id (str): Team ID (optional).
+
+        Returns:
+        dict: Result of the check containing 'entity_exist' and 'messages'.
+    """
+
+    llm = SherpaOpenAI(
+        temperature=cfg.TEMPRATURE,
+        openai_api_key=cfg.OPENAI_API_KEY,
+        user_id=user_id,
+        team_id=team_id,
+    )
+
+    instruction = f"""
+        I have a question and an answer. I want you to confirm if the entities from the question has been in some form inside the answer. 
+        Answer = {result}
+        Question = {source}
+        Entities inside the question = {source_entity}
+       """
+    instruction = (
+        instruction
+        + """
+           only return { entity_exist: 'true' , messages:"" } if entities are mentioned inside the answer in  
+           only return {entity_exist: 'false' , messages: " Entity x hasn't been mentioned inside the answer"} if the entity is not mentioned properly .
+          """
+    )
+
+    llm_result = llm.predict(instruction)
+
+    checkup_json = json_extractor(llm_result)
+    return checkup_json
+
+
+def text_similarity_by_metrics(check_entity: List[str], source_entity: List[str]):
+    """
+        Check entity similarity based on Jaccard and Levenshtein metrics.
+
+        Args:
+        - check_entity (List[str]): List of entities to check.
+        - source_entity (List[str]): List of reference entities.
+
+        Returns:
+        dict: Result of the check containing 'entity_exist' and 'messages'.
+    """
+
+    threshold = 0.75
+    error_entity = []
+    message = ""
+    levenshtein_constant = 0.5
+    for entity in source_entity:
+        metrics_value = 0
+        for sub_entitiy in check_entity:
+            word1 = entity
+            word2 = sub_entitiy
+
+            combined_similarity = string_comparison_with_jaccard_and_levenshtein(
+                word1=word1, word2=word2, levenshtein_constant=levenshtein_constant
+            )
+            if metrics_value < combined_similarity:
+                metrics_value = combined_similarity
+        if metrics_value < threshold:
+            error_entity.append(entity)
+
+    if len(error_entity) > 0:
+        for entity in error_entity:
```

Comment:
entities = ", ".join(error_entity)
message = f"remember to address these entities {entities} in final the answer."